photutils/02_photutils_source_detection.ipynb

```diff
@@ -0,0 +1,1034 @@
+    "\n",
+    "(Generally, the best package for web queries of astronomical data is `astroquery`; however, the dataset we are using is a High Level Science Product (HLSP) and thus is not located within a catalog that could be queried with `astroquery`.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'https://archive.stsci.edu/pub/hlsp/xdf/hlsp_xdf_hst_acswfc-60mas_hudf_f435w_v1_sci.fits'\n",
+    "with fits.open(url) as hdulist:\n",
+    "    hdulist.info()\n",
+    "    data = hdulist[0].data\n",
+    "    header = hdulist[0].header"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As explained in the [previous notebook](01_photutils_background_estimation.ipynb) on background estimation, it is important to **mask** these data, as a large portion of the values are equal to zero. We will mask out the non-data, so all of those pixels that have a value of zero don't interfere with our statistics and analyses of the data. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the mask\n",
+    "mask = data == 0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Throughout this notebook, we are going to store our images in Python using a `CCDData` object (see [Astropy documentation](http://docs.astropy.org/en/stable/nddata/index.html#ccddata-class-for-images)), which contains a `numpy` array in addition to metadata such as uncertainty, masks, or units. In this case, our data is in electrons (counts) per second."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "unit = u.ct / u.s\n",
+    "xdf_image = CCDData(data, unit=unit, meta=header, mask=mask)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's look at the data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up the figure with subplots\n",
+    "fig, ax1 = plt.subplots(1, 1, figsize=(8, 8))\n",
+    "\n",
+    "# Plot the data\n",
+    "norm_image = ImageNormalize(vmin=1e-4, vmax=5e-2, stretch=LogStretch())\n",
+    "xdf_image_clipped = np.clip(xdf_image, 1e-4, None) # clip to plot with logarithmic stretch\n",
+    "fitsplot = ax1.imshow(np.ma.masked_where(xdf_image.mask, xdf_image_clipped), norm=norm_image)\n",
+    "\n",
+    "# Define the colorbar and fix the labels\n",
+    "cbar = plt.colorbar(fitsplot, fraction=0.046, pad=0.04, ticks=LogLocator(subs=range(10)))\n",
+    "labels = ['$10^{-4}$'] + [''] * 8 + ['$10^{-3}$'] + [''] * 8 + ['$10^{-2}$']\n",
+    "cbar.ax.set_yticklabels(labels)\n",
+    "\n",
+    "# Define labels\n",
+    "cbar.set_label(r'Flux Count Rate ($e^{-1}/s$)', rotation=270, labelpad=30)\n",
```

Comment:
It might be neat to use the unit itself to render the latex.  That is, you could do something like:
```suggestion
    "cbar.set_label(r'Flux Count Rate (${}$)'.format(xdf_image.unit.to_string('latex')), rotation=270, labelpad=30)\n",
```

Reply:
(same comment applies below in several places)